### ISSUE TITLE
OpenGL/Vulkan back-ends rely on OpenCL for card list

### ISSUE THREAD

**Issue by baldurk** [GPUOpen-Tools/RGA#11]
Similar to #7 the OpenGL and Vulkan backends rely on OpenCL to enumerate a list of cards, and it fails on non-AMD hardware.

I fixed e.g. GL just by removing that code. If CL was initialised by being requested, that array would [already be filled out earlier](https://github.com/GPUOpen-Tools/RGA/blob/master/RadeonGPUAnalyzerBackend/src/beBackend.cpp#L102) and then just enumerating all cards the same as #7 did for DX. It gives the same behaviour as DX and I think this is safe, but I'm not sure if GL/CL are more tied together than D3D/CL were, so I didn't want to make this a pull request just yet.

If you think it's a good change I can make a PR for the GL and vulkan back-ends.

Note: Even with this change, the actual compilation subsequently fails on non-AMD, but it does so gracefully instead of just crashing the process. Also in future maybe the compilation will work on non-AMD :smile:.

```patch
diff --git a/RadeonGPUAnalyzerBackend/src/beBackend.cpp b/RadeonGPUAnalyzerBackend/src/beBackend.cpp
index 40edca7..074c14b 100644
--- a/RadeonGPUAnalyzerBackend/src/beBackend.cpp
+++ b/RadeonGPUAnalyzerBackend/src/beBackend.cpp
@@ -447,23 +447,6 @@ bool Backend::GetSupportedPublicDevices(std::set<std::string>& devices)
         ret = true;
         devices = m_supportedPublicDevices;
     }
-    else
-    {
-        // Get the supported public devices from the OpenCL backend.
-        if (m_beOpenCL == nullptr)
-        {
-            m_beOpenCL = new beProgramBuilderOpenCL;
-        }
-
-        beKA::beStatus rc = m_beOpenCL->Initialize();
-
-        if (rc == beStatus_SUCCESS)
-        {
-            // Retrieve the supported public devices from the OpenCL runtime.
-            m_beOpenCL->GetSupportedPublicDevices(devices);
-            ret = true;
-        }
-    }

     return ret;
 }
diff --git a/RadeonGPUAnalyzerBackend/src/beProgramBuilderOpenGL.cpp b/RadeonGPUAnalyzerBackend/src/beProgramBuilderOpenGL.cpp
index 9887403..076c03b 100644
--- a/RadeonGPUAnalyzerBackend/src/beProgramBuilderOpenGL.cpp
+++ b/RadeonGPUAnalyzerBackend/src/beProgramBuilderOpenGL.cpp
@@ -132,6 +132,12 @@ beKA::beStatus beProgramBuilderOpenGL::GetDeviceTable(std::vector<GDT_GfxCardInf
         }
     }

+    // If this is empty then odds are you are not using AMD hardware.
+    if(table.empty())
+    {
+      AMDTDeviceInfoUtils::Instance()->GetAllCards(table);
+    }
+
     std::sort(table.begin(), table.end(), beUtils::GfxCardInfoSortPredicate);

     if (!table.empty())
```

**Comment by AmitBM** [comment]
Hi baldurk,

Thank you for your post.
We are already working on removing the dependency of the OpenGL and Vulkan backends on the OpenCL runtime. The next release of RGA, which is expected to happen later this month, should include these changes.

**Comment by AmitBM** [comment]
Hi baldurk,

This issue should be resolved with the latest version, which was released today (RGA-1.1):
The DirectX, OpenGL and Vulkan modes no longer depend on the OpenCL runtime.
